src/lib.rs

```diff
@@ -39,15 +38,29 @@ pub trait CommandPtraceSpawn {
 
 impl CommandPtraceSpawn for Command {
     fn spawn_ptrace(&mut self) -> Result<Child> {
-        let child = self.before_exec(|| {
-            // Opt-in to ptrace.
-            ptrace(PTRACE_TRACEME, 0, ptr::null_mut(), ptr::null_mut())?;
-            Ok(())
-        }).spawn()?;
+        let child = unsafe {
+            self.pre_exec(|| {
+                // Opt-in to ptrace.
+                match ptrace::traceme() {
```

Comment:
N.B.: `ptrace::traceme()` is the replacement for `ptrace(PTRACE_TRACEME, ...)`. It no longer returns an `io::Result` (or something directly convertible to an `io::Result`), so I unrolled it a bit here: system errors (i.e., `errno` results) get propagated via `from_raw_os_error`, while everything else acts like the `waitpid` below and becomes an `io::ErrorKind::Other`.

Reply:
You could write this more concisely as `ptrace::traceme().map_err(|e| { // the err case you have below })` I'm not hung up on it enough to even use the review suggestion tool to rewrite it for you, so you can fix it in a followup if you're motivated or it can be what it is. :)